### ISSUE TITLE
Линейная одномерная регрессия

### ISSUE THREAD

**Issue by klods2023** [klods2023/klods#8]
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error
import joblib
import matplotlib.pyplot as plt

# Загрузка данных из файла Excel
data = pd.read_excel('Data_base.xlsx')
X = data['N'].values.reshape(-1, 1)
y = data['Temp']

# Создание и обучение модели линейной регрессии
model = LinearRegression()
model.fit(X, y)

# Сохранение обученной модели
joblib.dump(model, 'linear_regression_model.pkl')

# Загрузка новых данных из файла Excel
new_data = pd.read_excel('New_data.xlsx')
new_X = new_data['N'].values.reshape(-1, 1)

# Предсказание значений T для новых данных
predicted_T = model.predict(new_X)

# Вывод предсказанных значений T
print('Предсказанные значения T для новых данных:')
for i, value in enumerate(predicted_T):
    print(f'Object {i+1}: {value}')

# Расчет коэффициента детерминации (R^2)
r2 = r2_score(y, model.predict(X))
print(f'Коэффициент детерминации (R^2): {r2}')

# Расчет средней абсолютной ошибки (MAE)
mae = mean_absolute_error(y, model.predict(X))
print(f'Средняя абсолютная ошибка (MAE): {mae}')

# Расчет среднеквадратичной ошибки (MSE)
mse = mean_squared_error(y, model.predict(X))
print(f'Среднеквадратичная ошибка (MSE): {mse}')

# Отображение решения на графике
plt.scatter(X, y, color='blue', label='Actual')
plt.scatter(new_X, predicted_T, color='red', label='Predicted')
plt.xlabel('N')
plt.ylabel('Temp')
plt.legend()
plt.show()

**Comment by klods2023** [comment]
[Линейная одномерная регрессия.docx](https://github.com/klods2023/klods/files/13428657/default.docx)

**Comment by klods2023** [comment]
[Data_base.xlsx](https://github.com/klods2023/klods/files/13428729/Data_base.xlsx)
[New_data.xlsx](https://github.com/klods2023/klods/files/13428730/New_data.xlsx)

**Comment by klods2023** [comment]
Используется только 1 параметр объекта из таблицы
